### ISSUE TITLE
Change logic for weekly day-of-the-week checkbox visibility

### ISSUE THREAD

**Issue by romalytvynenko** [RebelCode/bookings-js#26]
### The Problem
When the end date is set to midnight of the next day, and repeating is set to "weeks", the day-of-the-week checkboxes do not show, which is technically wrong.

### Solution
Change the logic that decides when the checkboxes should change to be as follows:
```
get inputted start date, as S
calculate next date, as E', at 00:00
get inputted end date and time, as E
if E <= E', show checkboxes
```

**Comment by XedinUnknown** [comment]
The solution is good. But I don't understand why the problem is a problem. More specifically, why does special treatment need to be applied to this specific case.

**Comment by mecha** [comment]
When repeating with a "week" unit, the checkboxes for the days of the weeks are shown.
For instance: 15th June 2018 from 15:00 till 18:00, repeats every 1 week on Mon, Tue and Fri.

We had decided that giving the day of the week checkbox options didn't make a lot of sense when the range starts on different days, for instance 15th June 15:00 till 18th June 18:00. So we added a check to show the days of the weeks only when the range is on the same day.

So the problem is this: a user cannot set a range from 15th June 15:00 till the end of that day (15th June) **and** also repeat weekly for specific days of the week. Because to do so, they would need to set the range like this:
```
Starts: 15th June 2018 at 15:00
Ends: 16th June 2018 at 00:00
```
Since the range starts and ends of different _dates_, the options aren't shown. So the check in the code should not compare the dates (in the above example, 15 and 16). Instead, it should compare against _ending on a different date_. Ending on 16th June at midnight does not mean the range ends on a different date (since the end timestamp is exclusive).

My proposed algorithm above should cater for this, while not requiring the code to explicitly check _"if date is the day after but at midnight, then allow"_, because that would be a hard-coded exception to the rule, which is generally bad.
